### PR TITLE
Add secure etcd communication

### PR DESCRIPTION
**Pull request by jknipper**
This PR adds etcd certificates and configuration for secure communication with apiserver and backup sidecar. It is disabled by default for now.

### CODE REVIEW

File: charts/kube-master/templates/api.yaml [hunk 1]
```diff
@@ -165,6 +182,16 @@ spec:
...
+          livenessProbe:
+            exec:
+              command:
+                - sh
+                - -c
+                - cat < /dev/null > /dev/tcp/{{ include "etcd.fullname" . }}/2379
+            initialDelaySeconds: 60
+            periodSeconds: 5
+            timeoutSeconds: 1
+            failureThreshold: 1
```
Comment (jknipper) [inline charts/kube-master/templates/api.yaml]: @databus23 I also added a liveness probe to solve our cache problem if etcd is gone.

Comment (jknipper) [reply charts/kube-master/templates/api.yaml]: Moved this to a separate PR.

File: charts/kube-master/charts/etcd/templates/deployment.yaml [hunk 2]
```diff
@@ -43,11 +43,27 @@ data:
     exec etcd \
       --name=kubernikus \
       --data-dir=/var/lib/etcd/new.etcd \
+{{- if .Values.secure.enabled }}
+      --cert-file=/etc/kubernetes/certs/tls-etcd.pem \
+      --key-file=/etc/kubernetes/certs/tls-etcd-key.pem \
+      --client-cert-auth=true \
+      --trusted-ca-file=/etc/kubernetes/certs/etcd-clients-ca.pem \
+      --peer-cert-file=/etc/kubernetes/certs/tls-etcd.pem \
```
Comment (BugRoger) [inline charts/kube-master/charts/etcd/templates/deployment.yaml]: Shouldn't this be a certificate signed with the `etcd-peers-ca`?

File: pkg/apis/kubernikus/v1/secret.go [hunk 3]
```diff
@@ -103,6 +105,11 @@ type Certificates struct {
 	TLSWormholePrivateKey   string `json:"tls-wormhole-key.pem"`
 	TLSWormholeCertificate  string `json:"tls-wormhole.pem"`
 
+	TLSEtcdCAPrivateKey  string `json:"tls-etcd-ca-key.pem"`
```
Comment (BugRoger) [inline pkg/apis/kubernikus/v1/secret.go]: What is Etcd using TLS for?

File: pkg/apis/kubernikus/v1/secret.go [hunk 4]
```diff
@@ -103,6 +105,11 @@ type Certificates struct {
...
+	TLSEtcdCACertificate string `json:"tls-etcd-ca.pem"`
+	TLSEtcdPrivateKey    string `json:"tls-etcd-key.pem"`
```
Comment (BugRoger) [inline pkg/apis/kubernikus/v1/secret.go]: Shouldn't this be a `EtcdPeersPrivateKey`? Do we even have etcd peers? 🤣

**Comment by databus23** [comment]
go away stale bot

**Comment by jknipper** [comment]
@databus23 I rebased this on current master, looks good afaik.

**Comment by jknipper** [comment]
CA and cert generation has been moved to separate PRs #502 #515

File: charts/kube-master/charts/etcd/templates/deployment.yaml (hunk 2 above)
Comment (databus23) [reply charts/kube-master/charts/etcd/templates/deployment.yaml]: @jknipper @BugRoger has a point here I think. we should use a different CA for peer communication. As we don't use more then one etcd I also think the whole peer configuration can be omitted, no?

**Comment by jknipper** [comment]
@databus23 I removed all peer settings from deployment. Please have a look.

File: pkg/apis/kubernikus/v1/secret.go (hunk 4 above)
Comment (databus23) [reply pkg/apis/kubernikus/v1/secret.go]: This is the server ca for client taking to etcd. its fine. :)

File: pkg/apis/kubernikus/v1/secret.go (hunk 3 above)
Comment (databus23) [reply pkg/apis/kubernikus/v1/secret.go]: for encryption I guess.

**Comment by databus23** [comment]
@BugRoger good to go from my side. from your end as well?

**Comment by BugRoger** [comment]
LGTM

**Pull request merged**